spec/tapioca/compilers/dsl/active_record_associations_spec.rb

```diff
@@ -407,7 +407,42 @@ def commenters_attributes=(attributes); end
     end
 
     describe("with errors") do
-      it("generates RBI file for broken has_many :through collection association with errors") do
+      it("generates RBI file for broken has_one association") do
+        add_ruby_file("schema.rb", <<~RUBY)
+          ActiveRecord::Migration.suppress_messages do
+            ActiveRecord::Schema.define do
+              create_table :posts do |t|
+                t.references(:author)
+              end
+            end
+          end
+        RUBY
+
+        add_ruby_file("post.rb", <<~RUBY)
+          class Post < ActiveRecord::Base
+            has_one :author
+          end
+        RUBY
+
+        expected = <<~RBI
+          # typed: strong
+
+          class Post
+            include GeneratedAssociationMethods
+
+            module GeneratedAssociationMethods; end
+          end
+        RBI
+
+        assert_equal(expected, rbi_for(:Post))
+
+        assert_equal(1, generated_errors.size)
+        assert_equal(<<~MSG, generated_errors.first)
+          Cannot generate association `author` on `Post` since the constant `Author` does not exist.
```

Comment:
Or even better if we can get the location where this association is defined?

Reply:
> Is it possible to include the type of association in the message? Something like
> 
> ```
> Cannot generate association `has_one: :author` on `Post` since the constant `Author` does not exist.
> ```

We don't have that specific information when the error is raised. It's failing before we get that far along because we're attempting to build the association via reflection. The code actually blows up when trying to determine that relation in https://github.com/Shopify/tapioca/blob/87bef08e088ea3b7496283dbf866b98425f6bebd/lib/tapioca/compilers/dsl/active_record_associations.rb#L172
The `type_for` method calls `relflection.klass` trying to build the association, and that's where the `NameError` gets raised.

> Or even better if we can get the location where this association is defined?

I don't intend to put it into the error message, but that was the idea behind the updated `Tapioca::Error` class, to print that information, and the error message together rather than jumbled together in error messages.